js/src/elements/star-rating.ts

```diff
@@ -0,0 +1,283 @@
+			</svg>
+		</label>
+
+		<input part="visually-hidden" type="radio" value="5" id="radio-5" />
+		<label for="radio-5">
+			<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16" style="overflow: visible" >
+				<path d="M7.1939 2.10167C7.52403 1.43275 8.47789 1.43274 8.80802 2.10167L10.3291 5.18372L13.7304 5.67795C14.4685 5.78522 14.7633 6.69239 14.2291 7.21307L11.768 9.61212L12.349 12.9996C12.4751 13.7348 11.7034 14.2955 11.0431 13.9484L8.00096 12.349L4.95879 13.9484C4.29853 14.2955 3.52684 13.7348 3.65294 12.9996L4.23394 9.61212L1.77277 7.21307C1.23861 6.69239 1.53336 5.78522 2.27156 5.67795L5.67281 5.18372L7.1939 2.10167Z"/>
+			</svg>
+		</label>
+
+		<span id="alert">
+			<label for="radio-1"><span id="label-1"><slot name="label-1"></slot></span></label>
+			<label for="radio-2"><span id="label-2"><slot name="label-2"></slot></span></label>
+			<label for="radio-3"><span id="label-3"><slot name="label-3"></slot></span></label>
+			<label for="radio-4"><span id="label-4"><slot name="label-4"></slot></span></label>
+			<label for="radio-5"><span id="label-5"><slot name="label-5"></slot></span></label>
+		</span>
+	</div>
+</fieldset>
+`;
+
+const template = starRatingTemplate;
+
+class StarRatingElement extends HTMLElement {
+	static get observedAttributes() {
+		return ['disabled', 'name', 'value'];
+	}
+
+	constructor() {
+		super();
+		this.attachShadow({ mode: 'open' });
+		this.shadowRoot?.appendChild(template.content.cloneNode(true));
+	}
+
+	get disabled() {
+		return this.hasAttribute('disabled');
+	}
+
+	set disabled(val) {
+		this.setAttribute('disabled', val.toString());
+		const fieldset = this.shadowRoot?.querySelector('fieldset') as HTMLFieldSetElement;
+		fieldset.disabled = val;
+		fieldset.querySelectorAll('input').forEach(input => (input.disabled = val));
+	}
+
+	get name() {
+		return this.getAttribute('name') ?? '';
+	}
+
+	set name(value) {
+		this.setAttribute('name', value);
+	}
+
+	get value() {
+		return Math.max(0, Math.min(parseInt(this.getAttribute('value') || '0'), 5));
+	}
+
+	set value(val) {
+		this.setAttribute('value', val.toString());
+	}
+
+	/** TODO: add form related properties */
+
+	connectedCallback() {
+		this.shadowRoot?.addEventListener('change', this);
+		// arrow function to bind `this` value to star rating in handleFormData
+		this.closest('form')?.addEventListener('formdata', this);
+		this.shadowRoot?.querySelectorAll('input[type="radio"]').forEach(input => {
+			/* Upon initial focus on a brand new star rating, set value to match the currently focused star.
+			Otherwise, the value remains at 0 until user selects the focused star. */
```

Comment:
I think it would be best to align with standard radio behavior and not force selection here.

Reply:
removed